src/libcore/fmt/mod.rs

```diff
@@ -1124,6 +1124,39 @@ impl<T: Debug> Debug for [T] {
     }
 }
 
+macro_rules! fmt_dst {
+    ($($Trait:ident => $fmt_char:expr),*) => {
+        $(
+            impl<T: $Trait> $Trait for [T] {
+                fn fmt(&self, f: &mut Formatter) -> Result {
+                    try!(write!(f, "["));
+                    let mut is_first = true;
+                    for x in self.iter() {
+                        if is_first {
```

Comment:
This was how Debug was implemented for [T] before the builder stuff landed. I assumed it was because enumerate() was in std::iter and this is implemented in libcore. I'll check.